.github/workflows/gh-pages.yml

```diff
@@ -2,32 +2,50 @@ name: GitHub Pages
 
 on:
   push:
-    branches: [ master, audit, dev ]
+    branches: [ master, dev, feat/revamp-docs ]
   pull_request:
 
-concurrency: 
+concurrency:
   group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
   cancel-in-progress: true
 
 jobs:
-
   publish:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
-      - name: Setup mdBook
-        uses: peaceiris/actions-mdbook@v1
+      - name: Use Node.js 20
+        uses: actions/setup-node@v3
         with:
-          mdbook-version: '0.4.18'
-
-      - name: Build
-        run: mdbook build
+          node-version: 20
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Run typedoc
+        run: |
+          npm install 
+          npm run bootstrap
+          npm run build
+          npm run docs
+
+      - name: Fix typedoc links 
+        run: |
+          npm install
+          npm run setup-typedoc
+        working-directory: website
+
+      - name: Build 🔧
+        run: |
+          npm run build
+        working-directory: website
+
+      - name: Add CNAME
+        run: |
+          echo 'maci.pse.dev' > ./website/build/CNAME
 
       - name: Deploy
         uses: peaceiris/actions-gh-pages@v3
-        if: ${{ github.ref == 'refs/heads/dev' }}
```

Comment:
Removing the `if` condition here will lead to deploying for all PRs, which might not be optimal. Should we revisit this to ensure deployments are only for specific branches or events?

Reply:
so it would not deploy on main but only on dev?

Reply:
I think it's ok. isn't it? We can maintain latest information solely from `dev` branch (including official(?) releases that would happen from main).

Reply:
oh yea then should we just keep as trigger push to dev?